packages/material-ui/src/Tooltip/useKeyboardFocus.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+
+/**
+ *
+ * @param {boolean} disabled
+ * @returns true if called during a keyboard event
+ */
+export default function useKeyboardFocus(disabled) {
```

Comment:
What about using https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/ButtonBase/focusVisible.js. I believe the goal is the same.

Reply:
> What about using /packages/material-ui/src/ButtonBase/focusVisible.js@next

> Applicable to ButtonBase? What is the difference between focus-visible and keyboard focus?


I didn't have time to look into the focusVisible more but it looks over-engineered for a simple keyboard focus detection. I believe focus visible is different from keyboard focus.

Reply:
> I believe focus visible is different from keyboard focus.

I would expect it to be the exact same thing. It's here to differentiate the UI response between a mouse and keyboard interaction.

Reply:
> I didn't have time to look into the focusVisible more but it looks over-engineered for a simple keyboard focus detection.

Happy to simplify it :)